sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/sns/PublishResultCoder.java

```diff
@@ -42,12 +50,27 @@ static PublishResultCoder of() {
   public void encode(PublishResult value, OutputStream outStream)
       throws CoderException, IOException {
     StringUtf8Coder.of().encode(value.getMessageId(), outStream);
+    StringUtf8Coder.of().encode(value.getSdkResponseMetadata().getRequestId(), outStream);
```

Comment:
Since `SdkResponseMetadata` and `SdkHttpMetadata` seem to be used by multiple parts of AWS SDK I was wondering if we could create specific Coders (and tests) for both objects and just reuse them here. I know this could add some extra chores to this PR but we may benefit from them eventually in other IOs..

Reply:
Agree with @iemejia